api/src/context.rs

```diff
@@ -900,17 +902,30 @@ impl Context {
         if !config.enabled {
             return Ok(self.default_gas_estimation(min_gas_unit_price));
         }
+        if let Some(static_override) = &config.static_override {
+            return Ok(GasEstimation {
+                deprioritized_gas_estimate: Some(static_override.low),
+                gas_estimate: static_override.market,
+                prioritized_gas_estimate: Some(static_override.aggressive),
+            });
+        }
 
         let epoch = ledger_info.epoch.0;
 
         // 0. (0) Return cached result if it exists
-        if let Some(cached_gas_estimation) = self.cached_gas_estimation(epoch) {
+        let cache = self.gas_estimation_cache.read().unwrap();
```

Comment:
Silly question: do you need this read lock here? `self.cached_gas_estimation(&cache, epoch)` internally grabs the read lock, right?

Reply:
Aah, missed that -- makes sense, thanks 😄